noir_stdlib/src/string.nr

```diff
@@ -0,0 +1,11 @@
+use crate::collections::vec::Vec;
+impl<N> str<N> {
+    /// return a byte slice of the str content
+    #[builtin(str_as_bytes)]
+    fn as_bytes(_self: Self) -> [u8] { }
```

Comment:
```suggestion
    /// Converts the given string into a byte array
    #[builtin(str_as_bytes)]
    fn as_bytes(_self: Self) -> [u8; N] { }
```
We should probably return an array, right? Since we know the size here

Reply:
We return slices for field to bytes methods, so this does follow that

Reply:
For converting field to bytes though we don't know how many bytes will be returned ahead of time

Reply:
Yeah I guess if we already have `N` due to the string size we may as well use it